beater/beater.go

```diff
@@ -730,6 +730,13 @@ func (s *serverRunner) newFinalBatchProcessor(
 		v.OnKey("total")
 		v.OnInt(stats.Added)
 	})
+	// TODO: Where do we want to put this metric?
+	monitoring.NewFunc(monitoring.Default, "libbeat.output", func(_ monitoring.Mode, v monitoring.Visitor) {
```

Comment:
I'm not sure this will behave as expected.

I had some problems adding metrics when the names of the metrics "clashed" with each other. For example, on line 716, we're adding `libbeat.output.events`, which creates the `libbeat.output` parent. Then on this line, we're adding `libbeat.output` again, which normally wipes the contents of what was previously in `libbeat.output` (`libbeat.output.events....`).

Could you extend the existing test `TestServerElasticsearchOutput` to assert the output of the registry and that it behaves as we expect?